plugins/test/unit/plugins/migrations/test_0016_new_yum_distributor.py

```diff
@@ -173,6 +174,28 @@ def test_re_publish_repository(self, mock_publish):
 
         mock_publish.assert_called_once()
 
+    def test_distributor_working_dir(self):
+        distributor_id = 'distid'
+        repo_id = 'repoid'
+        storage_dir = pulp_config.config.get('server', 'storage_dir')
+        actual_dir = '%s/working/repos/%s/distributors/%s' % (storage_dir, repo_id, distributor_id)
+        directory = self.migration_module.distributor_working_dir(distributor_id, repo_id,
+                                                                  mkdir=False)
+        self.assertEqual(actual_dir, directory)
+
+    @mock.patch('os.makedirs')
+    def test_distributor_working_dir_makedirs(self, mock_makedirs):
+        distributor_id = 'distid'
+        repo_id = 'repoid'
+        storage_dir = pulp_config.config.get('server', 'storage_dir')
+        repo_working_dir = os.path.join(storage_dir, 'working', 'repos', repo_id)
+        distributor_working_dir = '%s/working/repos/%s/distributors/%s' % (storage_dir, repo_id,
+                                                                           distributor_id)
+        directory = self.migration_module.distributor_working_dir(distributor_id, repo_id)
+        mock_makedirs.assert_has_calls([mock.call(repo_working_dir),
+                                        mock.call(distributor_working_dir)])
+        self.assertEqual(distributor_working_dir, directory)
```

Comment:
Since this test does not mock os.path.exists, it is possible that if someone creates a repo and distributor with these ids on their Pulp installation that this test could fail (and it would be surprising). You could add a mock that returns False. That said, those particular names seem unlikely for devs to use.